Sources_v3/Utils/Internet Connection/InternetConnection.swift

```diff
@@ -0,0 +1,250 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import Network
+
+/// A delegate to receive Internet connection events.
+protocol InternetConnectionDelegate: class {
+    /// Calls when the Internet connection status did change.
+    /// - Parameter status: an Internet connection status.
+    func internetConnectionStatusDidChange(status: InternetConnection.Status)
+    
+    /// Calls when the Internet connection monitor got an error.
+    /// For example `Reachability` didn't initiated or started notifying properly.
+    /// - Parameter error: an error.
+    func internetConnectionMonitorFailed(with error: Error)
+}
+
+/// A protocol for Internet connection monitors.
+protocol InternetConnectionMonitor {
+    /// A delegate for receiving Internet connection events.
+    var delegate: InternetConnectionDelegate? { get set }
+    
+    /// The current status of Internet connection.
+    var status: InternetConnection.Status { get }
+    
+    /// Start Internet connection monitoring.
+    func start()
+    /// Stop Internet connection monitoring.
+    func stop()
+}
```

Comment:
Do you think we need this abstraction? The `InternetConnection` itself is a wrapper and this is an additional wrapper around the wrapper. I'd be completely fine with having just the `InternetConnection` concrete type.

Reply:
I think so, to unify API for different monitors.